src/test/unit/engine/NetworkFactoryTest.cpp

```diff
@@ -0,0 +1,122 @@
+/* ---------------------------------------------------------------------
+ * Numenta Platform for Intelligent Computing (NuPIC)
+ * Copyright (C) 2013, Numenta, Inc.  Unless you have an agreement
+ * with Numenta, Inc., for a separate license for this software code, the
+ * following terms and conditions apply:
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses.
+ *
+ * http://numenta.org/licenses/
+ * ---------------------------------------------------------------------
+ */
+
+/** @file
+ * Implementation of Network test
+ */
+
+#include "gtest/gtest.h"
+
+#include <nupic/engine/NetworkFactory.hpp>
+#include <nupic/engine/Network.hpp>
+#include <nupic/engine/NuPIC.hpp>
+#include <nupic/engine/Region.hpp>
+#include <nupic/utils/Log.hpp>
+
+using namespace nupic;
+
+#define SHOULDFAIL_WITH_MESSAGE(statement, message) \
```

Comment:
Can you use one of the gtest assertions instead of defining this?

https://github.com/google/googletest/blob/master/googletest/docs/AdvancedGuide.md#exception-assertions